Test/Uninstall-WhiskeyNuGetPackage.Tests.ps1

```diff
@@ -22,86 +18,138 @@ function WhenUninstallingNuGetPackage
+        It 'should uninstall all NuGet package versions with the same name' {
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.OpenCover' -WithVersion '4.7.922'
+            WhenUninstallingNuGetPackage -WithName 'NUnit.Runners' -WithVersion ''
+            ThenNuGetPackageUnInstalled -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            ThenNuGetPackageUnInstalled -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageNotUninstalled -WithName 'NUnit.OpenCover' -WithVersion '4.7.922'
+            ThenRanSuccessfully
         }
     }
-    
-    Describe 'Uninstall-WhiskeyNuGetPackage.when given an NuGet Package with a wildcard Version' {
-        It 'should never have installed the package' {
-            GivenAnInstalledNuGetPackage -WithVersion '2.*' -ErrorAction SilentlyContinue
-            WhenUninstallingNuGetPackage -WithVersion '2.*' -ErrorAction SilentlyContinue
-            ThenNuGetPackageNotUnInstalled -WithVersion '2.*' -WithError 'Wildcards are not allowed for NuGet packages'
+
+    Describe 'Uninstall-WhiskeyNuGetPackage.when given a NuGet package with a wildcard Version' {
+        It 'should uninstall all NuGet package versions with the same name' {
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.OpenCover' -WithVersion '4.7.922'
+            WhenUninstallingNuGetPackage -WithName 'NUnit.Runners' -WithVersion '*'
+            ThenNuGetPackageUnInstalled -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            ThenNuGetPackageUnInstalled -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageNotUninstalled -WithName 'NUnit.OpenCover' -WithVersion '4.7.922'
+            ThenRanSuccessfully
         }
     }    
+
+    Describe 'Uninstall-WhiskeyNuGetPackage.when given a NuGet package with a pinned wildcard Version' {
+        It 'should uninstall all NuGet package versions with the same name' {
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            GivenAnInstalledNuGetPackage -WithName 'NUnit.Runners' -WithVersion '3.6.4'
+            WhenUninstallingNuGetPackage -WithName 'NUnit.Runners' -WithVersion '2.*'
+            ThenNuGetPackageUnInstalled -WithName 'NUnit.Runners' -WithVersion '2.6.3'
+            ThenNuGetPackageUnInstalled -WithName 'NUnit.Runners' -WithVersion '2.6.4'
+            ThenNuGetPackageNotUninstalled -WithName 'NUnit.Runners' -WithVersion '3.6.4'
+            ThenRanSuccessfully
+        }
+    }    

```

Comment:
These two tests are duplicates. It doesn't matter what the wildcard is, just that `Uninstall-WhiskeyNuGetPackage` uses wildcard versions. Remove the test that uses `*`.